olm-catalog/serverless-operator/project.yaml

```diff
@@ -68,9 +68,7 @@ dependencies:
 upgrade_sequence:
     - csv: serverless-operator.v1.29.0
       source: redhat-operators
-    - csv: serverless-operator.v1.29.1
-      source: redhat-operators
     - csv: serverless-operator.v1.30.0
-      source: serverless-operator
+      source: redhat-operators
```

Comment:
So, for previous versions we should always use `redhat-operators` or from 1.31+ we will use `serverless-operator`? I'm wondering if we need to change the bump automation here: https://github.com/openshift-knative/serverless-operator/blob/b2506aa104c5e78995354c5296bc44f6d771f7b2/hack/cmd/bumpso/bumpso.go#L78C2-L86

Reply:
`redhat-operator` is used for releases that are already live. `serverless-operator` is used for versions that were not yet released to public (so, they're still in our internal `serverless-operator` catalog source).
Sometimes this sequence will use `serverless-operator` just for the last version in the upgrade sequence. But sometimes even one before. That's for the period when the release branch was already cut and we're starting to develop new version on `main` branch but the previous version was not released yet.

Reply:
Maybe we could get rid of this "source" parameter. We could always try to find the given version in the `redhat-operators` catalogsource. And if it's not there only then switch to `serverless-operator` source.

Reply:
I'm testing a solution locally now.

Reply:
OK. Everything pushed. It works locally.